File: gamslib/sparta/sparta-scalar.py
#  LP written by GAMS Convert at 12/13/18 10:24:49
#  
#  Equation counts
#      Total        E        G        L        N        X        C        B
#         11       11        0        0        0        0        0        0
#  
#  Variable counts
#                   x        b        i      s1s      s2s       sc       si
#      Total     cont   binary  integer     sos1     sos2    scont     sint
#         51       51        0        0        0        0        0        0
#  FX      0        0        0        0        0        0        0        0
#  
#  Nonzero counts
#      Total    const       NL      DLL
#        130      130        0        0
# 
#  Reformulation has removed 1 variable and 1 equation


from pyomo.environ import *

model = m = ConcreteModel()


m.x1 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x2 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x3 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x4 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x5 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x6 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x7 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x8 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x9 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x10 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x11 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x12 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x13 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x14 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x15 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x16 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x17 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x18 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x19 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x20 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x21 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x22 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x23 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x24 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x25 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x26 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x27 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x28 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x29 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x30 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x31 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x32 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x33 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x34 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x35 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x36 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x37 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x38 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x39 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x40 = Var(within=Reals,bounds=(0,None),initialize=0)
m.x41 = Var(within=Reals,bounds=(5,None),initialize=5)
m.x42 = Var(within=Reals,bounds=(6,None),initialize=6)
m.x43 = Var(within=Reals,bounds=(7,None),initialize=7)
m.x44 = Var(within=Reals,bounds=(6,None),initialize=6)
m.x45 = Var(within=Reals,bounds=(4,None),initialize=4)
m.x46 = Var(within=Reals,bounds=(9,None),initialize=9)
m.x47 = Var(within=Reals,bounds=(8,None),initialize=8)
m.x48 = Var(within=Reals,bounds=(8,None),initialize=8)
m.x49 = Var(within=Reals,bounds=(6,None),initialize=6)
m.x50 = Var(within=Reals,bounds=(4,None),initialize=4)

m.obj = Objective(expr=   50*m.x1 + 85*m.x2 + 115*m.x3 + 143*m.x4 + 52.5*m.x5 + 89.25*m.x6 + 120.75*m.x7 + 150.15*m.x8
                        + 56*m.x9 + 95.2*m.x10 + 128.8*m.x11 + 160.16*m.x12 + 85.5*m.x13 + 145.35*m.x14 + 196.65*m.x15
                        + 244.53*m.x16 + 90*m.x17 + 153*m.x18 + 207*m.x19 + 257.4*m.x20 + 95*m.x21 + 161.5*m.x22
                        + 218.5*m.x23 + 271.7*m.x24 + 98.5*m.x25 + 167.45*m.x26 + 226.55*m.x27 + 281.71*m.x28
                        + 105*m.x29 + 178.5*m.x30 + 241.5*m.x31 + 300.3*m.x32 + 111*m.x33 + 188.7*m.x34 + 255.3*m.x35
                        + 317.46*m.x36 + 119*m.x37 + 202.3*m.x38 + 273.7*m.x39 + 340.34*m.x40, sense=minimize)

m.c2 = Constraint(expr= - m.x1 - m.x2 - m.x3 - m.x4 + m.x41 == 0)

m.c3 = Constraint(expr=   m.x1 - m.x5 - m.x6 - m.x7 - m.x8 - m.x41 + m.x42 == 0)

m.c4 = Constraint(expr=   m.x2 + m.x5 - m.x9 - m.x10 - m.x11 - m.x12 - m.x42 + m.x43 == 0)

m.c5 = Constraint(expr=   m.x3 + m.x6 + m.x9 - m.x13 - m.x14 - m.x15 - m.x16 - m.x43 + m.x44 == 0)

m.c6 = Constraint(expr=   m.x4 + m.x7 + m.x10 + m.x13 - m.x17 - m.x18 - m.x19 - m.x20 - m.x44 + m.x45 == 0)

m.c7 = Constraint(expr=   m.x8 + m.x11 + m.x14 + m.x17 - m.x21 - m.x22 - m.x23 - m.x24 - m.x45 + m.x46 == 0)

m.c8 = Constraint(expr=   m.x12 + m.x15 + m.x18 + m.x21 - m.x25 - m.x26 - m.x27 - m.x28 - m.x46 + m.x47 == 0)

m.c9 = Constraint(expr=   m.x16 + m.x19 + m.x22 + m.x25 - m.x29 - m.x30 - m.x31 - m.x32 - m.x47 + m.x48 == 0)

m.c10 = Constraint(expr=   m.x20 + m.x23 + m.x26 + m.x29 - m.x33 - m.x34 - m.x35 - m.x36 - m.x48 + m.x49 == 0)

m.c11 = Constraint(expr=   m.x24 + m.x27 + m.x30 + m.x33 - m.x37 - m.x38 - m.x39 - m.x40 - m.x49 + m.x50 == 0)

File: gamslib/sparta/sparta-scalar.jl
#  LP written by GAMS Convert at 10/11/20 12:25:10
#  
#  Equation counts
#      Total        E        G        L        N        X        C        B
#         11       11        0        0        0        0        0        0
#  
#  Variable counts
#                   x        b        i      s1s      s2s       sc       si
#      Total     cont   binary  integer     sos1     sos2    scont     sint
#         51       51        0        0        0        0        0        0
#  FX      0        0        0        0        0        0        0        0
#  
#  Nonzero counts
#      Total    const       NL      DLL
#        130      130        0        0
# 
#  Reformulation has removed 1 variable and 1 equation


using JuMP
using MathOptInterface
const MOI = MathOptInterface

model = m = Model()

@variable(m, 0 <= x1, start=0)
@variable(m, 0 <= x2, start=0)
@variable(m, 0 <= x3, start=0)
@variable(m, 0 <= x4, start=0)
@variable(m, 0 <= x5, start=0)
@variable(m, 0 <= x6, start=0)
@variable(m, 0 <= x7, start=0)
@variable(m, 0 <= x8, start=0)
@variable(m, 0 <= x9, start=0)
@variable(m, 0 <= x10, start=0)
@variable(m, 0 <= x11, start=0)
@variable(m, 0 <= x12, start=0)
@variable(m, 0 <= x13, start=0)
@variable(m, 0 <= x14, start=0)
@variable(m, 0 <= x15, start=0)
@variable(m, 0 <= x16, start=0)
@variable(m, 0 <= x17, start=0)
@variable(m, 0 <= x18, start=0)
@variable(m, 0 <= x19, start=0)
@variable(m, 0 <= x20, start=0)
@variable(m, 0 <= x21, start=0)
@variable(m, 0 <= x22, start=0)
@variable(m, 0 <= x23, start=0)
@variable(m, 0 <= x24, start=0)
@variable(m, 0 <= x25, start=0)
@variable(m, 0 <= x26, start=0)
@variable(m, 0 <= x27, start=0)
@variable(m, 0 <= x28, start=0)
@variable(m, 0 <= x29, start=0)
@variable(m, 0 <= x30, start=0)
@variable(m, 0 <= x31, start=0)
@variable(m, 0 <= x32, start=0)
@variable(m, 0 <= x33, start=0)
@variable(m, 0 <= x34, start=0)
@variable(m, 0 <= x35, start=0)
@variable(m, 0 <= x36, start=0)
@variable(m, 0 <= x37, start=0)
@variable(m, 0 <= x38, start=0)
@variable(m, 0 <= x39, start=0)
@variable(m, 0 <= x40, start=0)
@variable(m, 5 <= x41, start=5)
@variable(m, 6 <= x42, start=6)
@variable(m, 7 <= x43, start=7)
@variable(m, 6 <= x44, start=6)
@variable(m, 4 <= x45, start=4)
@variable(m, 9 <= x46, start=9)
@variable(m, 8 <= x47, start=8)
@variable(m, 8 <= x48, start=8)
@variable(m, 6 <= x49, start=6)
@variable(m, 4 <= x50, start=4)

@objective(m, Min, 50*x1 + 85*x2 + 115*x3 + 143*x4 + 52.5*x5 + 89.25*x6 + 120.75*x7 + 150.15*x8 + 56*x9 + 95.2*x10
     + 128.8*x11 + 160.16*x12 + 85.5*x13 + 145.35*x14 + 196.65*x15 + 244.53*x16 + 90*x17 + 153*x18 + 207*x19 + 257.4*x20
     + 95*x21 + 161.5*x22 + 218.5*x23 + 271.7*x24 + 98.5*x25 + 167.45*x26 + 226.55*x27 + 281.71*x28 + 105*x29
     + 178.5*x30 + 241.5*x31 + 300.3*x32 + 111*x33 + 188.7*x34 + 255.3*x35 + 317.46*x36 + 119*x37 + 202.3*x38
     + 273.7*x39 + 340.34*x40)

@constraint(m,  - x1 - x2 - x3 - x4 + x41 == 0)

@constraint(m, x1 - x5 - x6 - x7 - x8 - x41 + x42 == 0)

@constraint(m, x2 + x5 - x9 - x10 - x11 - x12 - x42 + x43 == 0)

@constraint(m, x3 + x6 + x9 - x13 - x14 - x15 - x16 - x43 + x44 == 0)

@constraint(m, x4 + x7 + x10 + x13 - x17 - x18 - x19 - x20 - x44 + x45 == 0)

@constraint(m, x8 + x11 + x14 + x17 - x21 - x22 - x23 - x24 - x45 + x46 == 0)

@constraint(m, x12 + x15 + x18 + x21 - x25 - x26 - x27 - x28 - x46 + x47 == 0)

@constraint(m, x16 + x19 + x22 + x25 - x29 - x30 - x31 - x32 - x47 + x48 == 0)

@constraint(m, x20 + x23 + x26 + x29 - x33 - x34 - x35 - x36 - x48 + x49 == 0)

@constraint(m, x24 + x27 + x30 + x33 - x37 - x38 - x39 - x40 - x49 + x50 == 0)
